navigation/CameraFlowNavigator.tsx

```diff
@@ -0,0 +1,65 @@
+import { useNavigation } from "@react-navigation/native";
+import React, { FunctionComponent, ReactElement, useState } from "react";
+import { View } from "../components/Themed";
+import CameraRecording from "../screens/CameraRecording";
+import LetsStartRecording from "../screens/LetsStartRecording";
+import PrepareFace from "../screens/PrepareFace";
+import { Text } from "../components/Themed";
+
+interface CameraFlowNavigatorProps<T> {
+  objects: T[];
+  overlayCreator: (object: T) => ReactElement;
+}
+
+enum State {
+  prepareFace,
+  letsStart,
+  recording,
+}
+
+function CameraFlowNavigator<T>({
+  objects,
+  overlayCreator,
+}: CameraFlowNavigatorProps<T>): ReactElement {
```

Comment:
Where do these props come from? Also, why does this need to be parametrized?

Reply:
See your comment below lol.